test/core/iomgr/error_test.cc

```diff
@@ -187,16 +187,6 @@ static void test_os_error() {
   GRPC_ERROR_UNREF(error);
 }
 
-static void test_special() {
```

Comment:
Why remove? This should still work I think

Reply:
well, originally, when the parent was GRPC_ERROR_NONE, grpc_error_add_child would create an actual error out of it  which basically said - no error. Note that, this new error is no longer equal to GRPC_ERROR_NONE. 
For this new error, the original grpc_error_add_child would add a reference to the child, making it work. 

With the new semantics, we won't be creating this new error. Instead, we would just return a reference to child

Reply:
From a quick survey, I did not find any code making use of that behavior which is why I removed it

Reply:
Won't a GRPC_ERROR_NONE get GRPC_STATUS_OK when `grpc_error_get_int` is called? 
https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/error.cc#L458

Reply:
Ack, good point